### PR TITLE
Handle null status events from devices

### PR DESCRIPTION
**Pull request by knolleary**
## Description

Devices can publish a `null` status update after then reconnect their MQTT connection (https://github.com/FlowFuse/device-agent/blob/93e7b488271c7241df4a6e42db80ef074c84458d/lib/mqtt.js#L66) - as this call is unguarded unlike other status checkins.

There is a question on whether the device should be doing that, but given its the current behaviour, we need to handle it.

This PR updates the code to deal with a null payload - inferring its an `updating` device - and skips any further checks. The device will finish updating and its regular checkin will resync its state.

### CODE REVIEW

File: forge/db/controllers/Device.js [hunk 1]
```diff
@@ -12,29 +12,35 @@ module.exports = {
...
         device.set('lastSeenAt', literal('CURRENT_TIMESTAMP'))
-        if (!state.snapshot || state.snapshot === '0') {
-            if (device.activeSnapshotId !== null) {
-                device.set('activeSnapshotId', null)
-            }
+        if (!state) {
+            // We have received a `null` state from the device. That means it
+            // is busy updating itself. Update out local state to infer as much
+            // as we can from that
+            device.set('state', 'updating')
```
Comment (Steve-Mcl) [inline forge/db/controllers/Device.js:20]:
```suggestion
        if (!state) {
            // We have received a `null` state from the device. That means it
            // is busy updating itself. Update our local state to infer as much
            // as we can from that
            device.set('state', 'updating')
```

Super minor typo (had to read comment several times)

**Review by Steve-Mcl — APPROVED** [review]
All good bar minor comment that tripped my brain!

**Comment by knolleary** [comment]
@Steve-Mcl for future reference, if you propose a change for a minor typo, you are encouraged to self-merge the proposal to get it fixed and then merge the PR as a whole - rather than wait for me to get back to this.

**Comment by Steve-Mcl** [comment]
> @Steve-Mcl for future reference, if you propose a change for a minor typo, you are encouraged to self-merge the proposal to get it fixed and then merge the PR as a whole - rather than wait for me to get back to this.

Noted

**Pull request merged**